### ISSUE TITLE
ConcurrentModificationException when calling notify(Throwable)

### ISSUE THREAD

**Issue by pmecho** [bugsnag/bugsnag-android#931]
We're seeing logs in our production app when calling `Client.notify` ever since switching to `5.0.0` (also seen on `5.0.1`)

Stacktrace:
```
java.util.ConcurrentModificationException
        at java.util.LinkedHashMap$LinkedHashIterator.nextEntry(LinkedHashMap.java:411)
        at java.util.LinkedHashMap$EntryIterator.next(LinkedHashMap.java:430)
        at java.util.LinkedHashMap$EntryIterator.next(LinkedHashMap.java:430)
        at com.bugsnag.android.Metadata.toMap(Metadata:155)
        at com.bugsnag.android.Metadata.copy(Metadata:147)
        at com.bugsnag.android.EventInternal.<init>(EventInternal:12)
        at com.bugsnag.android.Event.<init>(Event:33)
        at com.bugsnag.android.Client.notify(Client:615)
```

**Comment by johnkiely1** [comment]
Hi @pmecho,

Thanks for reporting this, we are investigating and will update you as soon as we know more.

**Comment by xander-jones** [comment]
Hey @pmecho, a fix for this has been released in https://github.com/bugsnag/bugsnag-android/releases/tag/v5.2.0

**Comment by pmecho** [comment]
@xander-jones We seem to bet getting a new stacktrace now:

```
Caused by: java.util.ConcurrentModificationException
        at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:757)
        at java.util.LinkedHashMap$LinkedKeyIterator.next(LinkedHashMap.java:780)
        at java.util.AbstractCollection.addAll(AbstractCollection.java:343)
        at java.util.HashSet.<init>(HashSet.java:119)
        at com.bugsnag.android.Metadata.addMetadata(Metadata:97)
        at com.bugsnag.android.Metadata.setRedactedKeys(Metadata:97)
        at com.bugsnag.android.Client.notifyInternal(Client:678)
        at com.bugsnag.android.Client.populateAndNotifyAndroidEvent(Client:660)
        at com.bugsnag.android.Client.notify(Client:616)
```

Not sure if this is related or not. Let me know if I should file a new issue.

**Comment by mattdyoung** [comment]
Hi @pmecho 

Thanks for reporting. That error has a different cause but the symptoms are the same so we've re-opened this issue and will release a further fix.

**Comment by phillipsam** [comment]
Hi @pmecho - we have now released a fix for this in [v5.2.2](https://github.com/bugsnag/bugsnag-android/releases/tag/v5.2.2
